### ISSUE TITLE
purl spec for OCI containers

### ISSUE THREAD

**Issue by nishakm** [package-url/purl-spec#85]
I noticed that an example of purl for containers looks like this:
```
pkg:docker/cassandra@sha256:244fd47e07d1004f0aed9c
pkg:docker/customer/dockerimage@sha256:244fd47e07d1004f0aed9c?repository_url=gcr.io
```
I'd like to address a few concerns with respect to containers in general

- Container images are not necessarily full packages but a collection of tarballs with some config files. So maybe the `scheme` isn't `pkg` for this "artifact" type?
- The `type` could be `docker` or `oci` but they can be one and the same based on how they are distributed and consumed. Is this something that can be chosen freely?

Maybe a good compromise would be something like this:
```
pkg:container/namespace/repo@sha256:244fd47e07d1004f0aed9c?domain_url=docker.io
```
What do you think?

**Comment by stevespringett** [comment]
Duplicate of #68

**Comment by iamwillbar** [comment]
Closing as this is being implemented and discussed in #123.
